doc/bucket-class-crd.md

```diff
@@ -1,73 +1,100 @@
+## Namespace Policy
+A namespace bucket class defines a policy for namespace buckets - i.e. NooBaa buckets that are backed by [namespacestores](namespace-store-crd.md).
+There are several types of namespace policies:
+- Single - a single namespace store is used for both read and write operations on the target bucket
+- Multi - a single namespace store is used for write operations, and a list of namespace stores can be used for read operations
+- Cache - functions similarly to `Single`, except with an additional `TTL` key, which dictates the time-to-live of the cached data
 
+### Time-to-live (TTL)
+Cache bucketclasses work by saving read objects in a chosen backingstore, which leads to faster access times in the future. In order to make sure that the cached object is not out of sync with the one in the remote target, an ETag comparison might be run upon read, depending on the TTL that the user chooses. The TTL can fall in one of three categories:
+- Negative (e.g. `-1`) - when the user knows there are no out of band writes, they can use a negative TTL, which means no revalidations are done; if the object is in the cache - it is returned without an ETag comparison. This is the most performant option.
+- Zero (`0`) - the cache will always compare the object's ETag before returning it. This option has a performance cost of getting the ETag from the remote target on each object read. This is the least performant option.
+- Positive (denoted in milliseconds, e.g. `3600000` equals to an hour) - once an object was read and saved in the cache, the chosen amount of time will have to pass prior to the object's ETag being compared again.
 
-Namespace policy of type single will require the following configuration:
-  - Resource - a single namespace-store, defines the read and write target of the namespace bucket.
+## Replication Policy
+It is possible to set a bucketclass-wide replication policy, that will be inherited and used by all future buckets created under that bucketclass.
 
-Namespace policy of type multi will require the following configuration:
-- Read Resources - list of namespace-stores, defines the read targets of the namespace bucket.
-- Write Resource - a single namespace-store, defines the write target of the namespace bucket.
+A replication policy is a JSON-compliant string which defines an array of rules -
+  - Each rule is an object containing a `rule_id`, a `destination bucket`, and an optional `filter` key that contains a `prefix` field.
+  - When a filter with prefix is provided - only objects keys that match the prefix will be replicated
 
-Namespace policy of type cache will require the following configuration:
-  - Hub Resource - a single namespace-store, defines the read and write target of the namespace bucket.
-  - TTL - defines the TTL of the cached data.
+Replication policy:
 
+A bucket-class will define a replication policy for all future namespace buckets who utilize it.
+The policy is a JSON-compliant array of rules.
```

Comment:
Would you please attach or link a full example of a replication policy as a JSON file?
It can be also from the tests and can appear below the examples and a note.